org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/serializer/SerializerFragment2.xtend

```diff
@@ -445,6 +446,9 @@ class SerializerFragment2 extends AbstractStubGeneratingFragment {
 		val clazz = if (isGenerateStub) grammar.abstractSyntacticSequencerClass else grammar.syntacticSequencerClass
 		val javaFile = fileAccessFactory.createGeneratedJavaFile(clazz)
 		javaFile.resourceSet = language.resourceSet
+    
+	    val elements = allAmbiguousTransitionsBySyntax
+	    val partitions = Iterables.partition(elements, 20)
```

Comment:
am not sure if the 20 is too agressing.
also some of the test languages are affected
@szarnekow what do you think

Reply:
It looks like being way too aggressive.
What is the largest number that would work with your use case?
Also can you provide a reasonably large grammar that illustrates the broken behavior? Also: What if that grammar is made even bigger: will the code still run into limits of the constant pool?

Reply:
Unfortunately I can't disclose the grammar. I retested it and the maximum number I could set it to was 63 so I updated my PR with 60. Currently our grammar (which we already seperated into multiple grammars) generates over 600 constants and yes if the grammar is even bigger it probably would run into a limit with the constants too.